ipa-core/src/test_fixture/world.rs

```diff
@@ -629,6 +629,9 @@ impl Runner<NotSharded> for TestWorld<NotSharded> {
                 let v = ctx.validator::<F>();
                 let m_ctx = v.context();
                 let r_share = m_ctx.clone().r(RecordId::FIRST).clone();
+
+                // Clippy complaining, but unclear why.
+                #[allow(clippy::disallowed_methods)]
```

Comment:
For some reason, without this, I get the [following compile error:](https://github.com/private-attribution/ipa/actions/runs/10566725128/job/29274134196#step:8:18)
```
error: use of a disallowed method `futures::future::join_all`
   --> ipa-core/src/test_fixture/world.rs:633:21
    |
633 | /                     join_all(zip(shares, iter::repeat(m_ctx.clone())).enumerate().map(
634 | |                         |(i, (share, m_ctx))| async move {
635 | |                             let record_id = RecordId::from(i);
636 | |                             let m_share = share.upgrade(m_ctx.clone(), record_id).await.unwrap();
...   |
644 | |                         },
645 | |                     ))
    | |______________________^
    |
    = note: We don't have a replacement for this method yet. Consider extending `SeqJoin` trait. (from clippy.toml)
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#disallowed_methods
    = note: `-D clippy::disallowed-methods` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::disallowed_methods)]`
```

It's not exactly clear why this change triggers this. I do import `TestWorld`, but I imported that when the file was in `ipa-core/cli/crypto.rs` as well.

Reply:
In general, we disallow the `join_all` method because we want to steer protocols to use `seq_join` where appropriate. `seq_join` is more careful about the number of futures it will spawn.

It's fine to use `join_all` here, though, because it's only three items, and it's test code. There is a general waiver of the `disallowed_methods` lint for `cfg(test)`, but not for `feature = test-fixture`.

So any of the following would be fine:
* Update the comment to say why it's okay to waive the lint (e.g. `// It's just 3 items.` appears elsewhere in world.rs).
* Use `join3` instead.
* Waive `disallowed_methods` whenever `feature = test-fixture`, or for all of `world.rs`, or for all of `mod test_fixture`.

Reply:
I'm still somewhat confused what code path all the sudden hits this (I didn't add the `join_all`.) It didn't seem to come up when I had it previously in the `src/cli/crypto.rs` file ¯\_(ツ)_/¯

I'll update the comment, presumably this will also come up when others use this code path.